internal/capacity/retrieve.go

```diff
@@ -72,39 +74,47 @@ func (c *Client) Retrieve() *cobra.Command {
 			if len(facilities) > 0 {
 				locationField = "Facility"
 				locs = append(locs, facilities...)
+
 			} else if len(metros) > 0 || checkMetro {
-				lister = c.Service.ListMetros
+				lister = metal.ApiFindCapacityForFacilityRequest(c.Service.FindCapacityForMetro(context.Background()))
 				locationField = "Metro"
 				locs = append(locs, metros...)
+
 			}
 
-			capacities, _, err := lister()
+			capacities, _, err := lister.Execute()
 			if err != nil {
 				return fmt.Errorf("Could not get Capacity: %w", err)
 			}
+			capacities.GetCapacity().ToMap()
 
 			header := []string{locationField, "Plan", "Level"}
 			data := [][]string{}
 
 			filtered := map[string]map[string]map[string]string{}
+			for AdditionalProperties, capacity := range *capacities {
 
-			for locCode, capacity := range *capacities {
-				if len(locs) > 0 && !contains(locs, locCode) {
-					continue
-				}
-				for plan, bm := range capacity {
-					if len(plans) > 0 && !contains(plans, plan) {
-						continue
-					}
-					loc := []string{locCode, plan, bm.Level}
-					data = append(data, loc)
-					if len(filtered[locCode]) == 0 {
-						filtered[locCode] = map[string]map[string]string{}
-					}
-					filtered[locCode][plan] = map[string]string{"levels": bm.Level}
-				}
 			}
 
+			// fmt.Println(capacityReport.ToMap())
+			fmt.Println(capacityReport)
+			// for capacity := range capacityReport {
```

Comment:
The API spec for capacities is in pretty bad shape; among other things, it has specific facility names hard-coded as property names.  We'll need to figure out a different schema structure before we can move forward with migrating the capacity commands to metal-go

Reply:
The `v0.10.0` release of `metal-go` is based off of an updated spec for capacities and should be suitable for converting these subcommands.  I recommend upgrading to `v0.10.0` in a separate PR, because you may need to update some commands that were already converted to `metal-go`.  For example, I think all of the `AdditionalProperties` calls in `cmd/init.go` will need to be replaced since the spec that was used for `v0.10.0` moves those specific properties out of `AdditionalProperties`.

Reply:
If you rebase on the latest from `main` you'll be able to revisit this command conversion.

Reply:
# packngo
```
vasu@vasu-cloud:~/Equinix/metal-cli$  metal capacity check -m ny,da -P c3.medium.x86 -q 10
+-------+---------------+----------+--------------+
| METRO |     PLAN      | QUANTITY | AVAILABILITY |
+-------+---------------+----------+--------------+
| ny    | c3.medium.x86 | 10       | true         |
| da    | c3.medium.x86 | 10       | true         |
+-------+---------------+----------+--------------+
```
# Metal-go
```
vasu@vasu-cloud:~/Equinix/metal-cli/bin$ ./metal capacity check -m ny,da -P c3.medium.x86 -q 10
+-------+---------------+----------+--------------+
| METRO |     PLAN      | QUANTITY | AVAILABILITY |
+-------+---------------+----------+--------------+
| da    | c3.medium.x86 | 10       | true         |
| da    | c3.medium.x86 | 10       | true         |
+-------+---------------+----------+--------------+
```
I verified with debug messages, I feel the code is OK. Can you check on your side too?

Reply:
The `metal-go` version of the output lists `da` twice instead of `ny` and `da`.

Reply:
That command would also make a good end-to-end test: run the subcommand with those flags and check that the data came back with the expected values in `metro` and `plan` (the other two fields are more likely to change between test runs so we can ignore them for now).

Reply:
Fixed in the latest commit.

Reply:
```
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal capacity check -m ny,da -P c3.medium.x86,m3.large.x86 -q 10
+-------+---------------+----------+--------------+
| METRO |     PLAN      | QUANTITY | AVAILABILITY |
+-------+---------------+----------+--------------+
| ny    | c3.medium.x86 | 10       | true         |
| ny    | m3.large.x86  | 10       | true         |
| da    | c3.medium.x86 | 10       | true         |
| da    | m3.large.x86  | 10       | true         |
+-------+---------------+----------+--------------+
Vasubabus-MacBook-Pro:bin vasubabu$ ./metal capacity check -m ny,da -P c3.medium.x86 -q 10
+-------+---------------+----------+--------------+
| METRO |     PLAN      | QUANTITY | AVAILABILITY |
+-------+---------------+----------+--------------+
| ny    | c3.medium.x86 | 10       | true         |
| da    | c3.medium.x86 | 10       | true         |
+-------+---------------+----------+--------------+
```